Search/Response+Parsing.swift

```diff
@@ -0,0 +1,154 @@
+//
+//  Response+Parsing.swift
+//  Search
+//
+//  Created by David Haynes on 02/03/2016.
+//  Copyright © 2016 Ordnance Survey. All rights reserved.
+//
+
+import Fetch
+import JSONLib
+
+public enum SearchError: ErrorType {
+    case NoDataReceived
+    case FailedToParseJSON
+    case FailedToDeserialiseJSON
+    case UnknownError
+}
+
+extension Response: Parsable {
+    public static func parse(fromData data: NSData?, withStatus status: Int) -> Result<Response> {
+        switch status {
+        case 200:
+            return parseResponse(data)
+        default:
+            return .Failure(SearchError.UnknownError)
+        }
+    }
+}
+
+private func parseResponse(data: NSData?) -> Result<Response> {
+    guard let data = data else {
+        return .Failure(SearchError.NoDataReceived)
+    }
+    guard let json = JSValue.parse(data).value else {
+        return .Failure(SearchError.FailedToParseJSON)
+    }
+    guard let response = responseFromJson(json) else {
+        return .Failure(SearchError.FailedToDeserialiseJSON)
+    }
+    return .Success(response)
+}
+
+private func responseFromJson(json: JSON) -> Response? {
+    let headerJson = json[Response.HeaderKey]
+    guard let header = headerFromJson(headerJson) else {
+        return nil
+    }
+    let resultsJson = json[Response.ResultsKey]
+    guard let results = resultsFromJson(resultsJson) else {
+        return nil
+    }
+    return Response(results: results, header: header)
+}
+
+private func headerFromJson(json: JSON) -> Header? {
+    guard let uri = json[Header.UriKey].string,
+        lr = json[Header.LrKey].string,
+        dataset = json[Header.DatasetKey].string,
+        outputSrs = json[Header.OutputSrsKey].string,
+        epoch = json[Header.EpochKey].string,
+        format = json[Header.FormatKey].string,
+        query = json[Header.QueryKey].string,
+        matchPrecision =  json[Header.MatchprecisionKey].number,
+        maxresults = json[Header.MaxresultsKey].number,
+        totalresults = json[Header.TotalresultsKey].number,
+        offset = json[Header.OffsetKey].number
+    else {
+        return nil
+    }
+
+    return Header(
```

Comment:
I was thinking when writing similar in the routing API, that maybe this stuff could be tidied up by having a nullable initialiser on the model object. That way the decision on whether to construct the object is taken by the class rather than this parsing file. Basically, move this stuff inside an `init?` method. That would make it more likely the mapping keys could then be made private.